rebound/librebound.py

```diff
@@ -172,7 +173,85 @@ def force_is_velocitydependent(self,value):
             return
         raise ValueError("Expecting integer.")
     
+# Units
 
+    @property
+    def units(self):
+        return self._units
+
+    @units.setter
+    def units(self, newunits):
+        newunits = self.check_units(newunits)        
+        if self.particles: # some particles are loaded
+            raise Exception("Error:  You must initialize the units before populating the particles array.  See Units.ipynb in python_tutorials.")
+        self.update_units(newunits) 
+
+    def convert_particle_units(self, *args): 
+        new_l, new_t, new_m = self.check_units(args)
+        for p in self.particles:
+            self.convert_p(p, self._units['length'], self._units['time'], self._units['mass'], new_l, new_t, new_m)
+        self.update_units((new_l, new_t, new_m))
+
+    def convert_p(self, p, old_l, old_t, old_m, new_l, new_t, new_m):
+        p.m = self.convert_mass(p.m, old_m, new_m)
+        p.x = self.convert_length(p.x, old_l, new_l) 
+        p.y = self.convert_length(p.y, old_l, new_l)
+        p.z = self.convert_length(p.z, old_l, new_l)
+        p.vx = self.convert_vel(p.vx, old_l, old_t, new_l, new_t)
+        p.vy = self.convert_vel(p.vy, old_l, old_t, new_l, new_t)
+        p.vz = self.convert_vel(p.vz, old_l, old_t, new_l, new_t)
+        p.ax = self.convert_vel(p.ax, old_l, old_t, new_l, new_t)
+        p.ay = self.convert_vel(p.ay, old_l, old_t, new_l, new_t)
+        p.az = self.convert_vel(p.az, old_l, old_t, new_l, new_t)
+        return p
+
+    def check_units(self, newunits):   
+        if len(newunits) is not 3:
+            raise Exception("Error: Need to pass exactly 3 units for length, time, and mass (any order), see python_tutorials/Units.ipynb")
+        
+        l_unit = t_unit = m_unit = None
+        for unit in newunits:
+            unit = unit.lower()
+            if unit in lengths_SI:
+                l_unit = unit
+            if unit in times_SI:
+                t_unit = unit
+            if unit in masses_SI:
+                m_unit = unit
+
+        if l_unit is None or t_unit is None or m_unit is None:
+            raise Exception("Error: Need to assign rebound.units a tuple consisting of 3 units for length, time, and mass (any order).  See python_tutorials/Units.ipynb.  If you did this, your unit isn't in our list.  Please update the dictionaries at the bottom of rebound/rebound/librebound.py and send a pull request!")
```

Comment:
I was just trying to explain that you could get this error if a) you didn't pass things in the right format, or b) if one of the units you passed weren't in our dictionary.  I've tried to rephrase to make more clear